.ci/Makefile

```
@@ -19,7 +19,7 @@ VAULT_TOKEN = $(shell vault write -address=$(VAULT_ADDR) -field=token auth/appro
 else
 VAULT_TOKEN = $(shell vault write -address=$(VAULT_ADDR) -field=token auth/github/login token=$(GITHUB_TOKEN))
 # we use roleId as a string that has to be there for authn/z for CI, but it's empty and not needed for local execution
-NOT_USED := $(shell test -e $(ROOT_DIR)/deployer-config.yml && sed -i -e "s;roleId:;token: $(GITHUB_TOKEN);g" $(ROOT_DIR)/deployer-config.yml)
+NOT_USED := $(shell test -e $(ROOT_DIR)/deployer-config.yml && sed -i '' -e "s;roleId: dev;token: $(GITHUB_TOKEN);g" $(ROOT_DIR)/deployer-config.yml)
```

I think `''` makes `sed` believe this is the path of an empty file for me.

```sh
> sed --version | head -1
sed (GNU sed) 4.4
```

```suggestion
NOT_USED := $(shell test -e $(ROOT_DIR)/deployer-config.yml && sed -i -e "s;roleId:.*;token: $(GITHUB_TOKEN);g" $(ROOT_DIR)/deployer-config.yml)
```

---

According to [stackoverflow](https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux), the only way to have a sed in-place command that works both for the GNU and the BSD version of `sed` is to set a dot prefix without space:

```suggestion
NOT_USED := $(shell test -e $(ROOT_DIR)/deployer-config.yml && sed -i.bak "s;roleId:.*;token: $(GITHUB_TOKEN);g" $(ROOT_DIR)/deployer-config.yml)
```

Otherwise, I'm also fine with using an intermediate temporary file:
 `sed ... > tmp.yml && mv tmp.yml > deployer-config.yml`.

```suggestion
NOT_USED := $(shell test -e $(ROOT_DIR)/deployer-config.yml && sed "s;roleId:.*;token: $(GITHUB_TOKEN);g" > tmp.yml && mv tmp.yml > $(ROOT_DIR)/deployer-config.yml)
```


Also, I think we can safely remove the `-e` as we only use one expression and search for `roleId: .*` instead of `roleId: dev` to be more radical.

---

Behavior differs between GNU and BSD seds, so moving to simpler sed invocation and using temp file.